pytables.ipynb

```diff
@@ -0,0 +1,417 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Comparison of disk storage methods\n",
+    "\n",
+    "This notebook is used to compare PyTables, an HDF5 Python implementation, with Numpy's native .npy/.npz file format"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import tables\n",
+    "\n",
+    "import pandas as pd\n",
+    "from neo4j.v1 import GraphDatabase\n",
+    "import hetio.readwrite\n",
+    "import hetio.neo4j\n",
+    "import hetio.pathtools\n",
+    "import numpy as np\n",
+    "from scipy import sparse\n",
+    "import time\n",
+    "from hetmech.matrix import get_node_to_position, metaedge_to_adjacency_matrix"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "url = 'https://github.com/dhimmel/hetionet/raw/76550e6c93fbe92124edc71725e8c7dd4ca8b1f5/hetnet/json/hetionet-v1.0.json.bz2'\n",
+    "graph = hetio.readwrite.read_graph(url)\n",
+    "metagraph = graph.metagraph"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# NPY Saving and loading"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "r, c, gig_adj = metaedge_to_adjacency_matrix(graph, 'GiG')\n",
+    "r, c, ctd_adj = metaedge_to_adjacency_matrix(graph, 'CtD')\n",
+    "r, c, sparse_gig_adj = metaedge_to_adjacency_matrix(graph, 'GiG', sparse_threshold=1)\n",
+    "r, c, sparse_ctd_adj = metaedge_to_adjacency_matrix(graph, 'CtD', sparse_threshold=1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Save matrices"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### np.save"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "645.2720165252686 ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Dense matrices into .npy\n",
+    "t1 = time.time()\n",
+    "np.save('data/gig_adj', gig_adj)\n",
```

Comment:
Let's add `.npy` to `file` argument. Numpy adds it automatically if missing, but that was a bit unexpected to me. I don't really like the idea of the program changing the file name. Also applies to `.npz` below.

Reply:
There seems to be an odd naming collision between scipy and numpy. In numpy, `.npy` is for saving a single array and `.npz` stores dictionary of arrays. However, scipy saves single arrays with the `.npz` extension. Confusing.